src/emitters/spot.cpp

```diff
@@ -57,7 +57,23 @@ class SpotLight final : public Emitter<Float, Spectrum> {
     MTS_IMPORT_TYPES(Scene, Texture)
 
     SpotLight(const Properties &props) : Base(props) {
-        m_intensity = props.texture<Texture>("intensity", Texture::D65(1.f));
+        if (props.has_property("intensity") && props.has_property("texture")) {
+            m_intensity = props.texture<Texture>("intensity");
+            m_texture = props.texture<Texture>("texture");
+            
+            if (m_intensity->is_spatially_varying())
```

Comment:
Actually, I think if we keep the two parameters, then `intensity` should never be spatially varying, even when `texture` isn't specified.

Reply:
Should texture be always varying even if intensity is not specified?

Reply:
Yes I would say so